src/gui.rs

```diff
@@ -16,6 +16,9 @@ pub struct SATApp {
     constraints: ConstraintList,
     callback_wrapper: CadicalCallbackWrapper,
     solver: Solver<CadicalCallbackWrapper>,
+    rendered_constraints: Vec<Vec<i32>>,
```

Comment:
Could this be a vector of indexes into the actual constainttlist, since we have already allocated space for them.

Reply:
Would it cause any unpredictable changes when using other functions, how would you estimate? If not, possibly worth trying. Comments from actual Rustaceans? ;-)

Reply:
Yes, it could, but I'm unsure if the added complexity is worth it. There are no problems with the performance currently, and the saved memory would be a few megabytes.

Reply:
We are going to need to know which constraint was clicked (and be able to communicate that info to another part of the code) very soon in development. Can the current architecture support that in a reasonable way?

Reply:
> We are going to need to know which constraint was clicked (and be able to communicate that info to another part of the code) very soon in development. Can the current architecture support that in a reasonable way?

Yeah that should be doable with the current setup

Reply:
I'm closing this conversation since at the moment there is no need for the change. If in the future the cloning gets too slow we can revisit this. Maybe if we start to support variable sized sudokus?